### PR TITLE
Private_events

### PR DESCRIPTION
**Pull request by jacobo12montes**
Kindly assist in reviewing our project again. The corrections have been implemented based on the previous TSEs' recommendations. Thank you.

### CODE REVIEW

**Review by javitocor — CHANGES_REQUESTED** [review]
# STATUS CHANGES REQUESTED ♻️ 

Good effort in this project, team!
Your project looks really good! 👍 
WE just need to implement some small changes and ready to go 💪 
Kindly check my comments below 👇 implement the changes and resubmit the project for a new code review ⏩ 
Good luck! 😺 
## Mandatory changes
- Add routes only for used actions, i.e. resources :events, only: [:new, :create, :show, :index]
- In the users controller, show actions must be like:
* [ ] lists events created by user, i.e. `@created_events = current_user.events` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/current_user_show_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [ ] lists events previoulsy attendented by user **using event model scope**, i.e. `@past_events = current_user.attended_events.previous` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
* [ ] lists upcoming events previously attended by user **using event model scope**, i.e. `@upcoming_events = current_user.attended_events.upcoming` ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/user_events_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations))
- A user should not be able to delete or update events created by other users
- Fix your IU, on the login and sign up pages there are many links pointing to the same. ie 2 buttons for login and 2 buttons for sign up. Kindly fix it, you can add some logic to your views.
- In your readme file, add the name of the Twitter, Linkedin and GitHub profile, now looks like:
Gmail: @gmail
Github: @githubhandle
Twitter: @twitterhandle
Linkedin: linkedin
- Kindly, add tests to your app, now there are no tests available
- Add foreign keys on your database

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__You've made laudable progress in this project.
However, some changes are required before this project can be approved. Please check my comments below and fix them as appropriate.__

**[ REQUIRED CHANGES ]**

- [ ] **You have some work to do concerning your test cases**
- Add more test cases: you added test cases for user registration and creating a new event. You can add more test cases to attend an event, show all events, etc.
  - Your models should have unit tests for validations and associations (Rspec) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/unit_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
  - Authentication have has integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))
  - Event management features have integration tests (Rspec + Capybara) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/associations/integration_tests_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49723079/))

- [ ] **You should include the instruction on how to set up and run your test cases to your visitors in your README.md file.** 

**[ COMMENT ]**

This project is built with ruby version 2.5.1 This is relatively old. I will recommend you update your ruby version to probably 2.7.1 or 2.7.2 in your subsequent project.

* * *
**Please make the required changes and submit another review request when you are done.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Comment by AdesojiCodeMaster** [comment]
Please, assist in reviewing our project again. All the corrections have been implemented based on the previous TSEs' recommendations. Thank you.

File: spec/models/event_attendance_spec.rb
Comment (addod19) [inline spec/models/event_attendance_spec.rb:12]: This part of the code is failing. Let us kindly address this. The error is "expected `true` but got `false`" :100:

File: spec/features/user_spec.rb
Comment (addod19) [inline spec/features/user_spec.rb]: This part of the code is failing. Let us kindly address it :slightly_smiling_face:

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi @AdesojiCodeMaster and @jacobo12montes, congratulations on fixing most of the comments made by the previous TSE's :fireworks: The only issue we have now is the failing test cases as outlined in the comments below: :smile: 
Kindly fix this and submit another core review :man_technologist: 

_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Comment by AdesojiCodeMaster** [comment]
Please, the recommendations have been addressed according to the TSE comments. Thank you.

**Review by krys2fa — APPROVED** [review]
# Status: Project Approved! 🥇 
Congratulations on making all the requested changes and getting this project approved. 👍 

You can now merge this branch into the development branch and go on with your next project.

💡  Do not forget to submit your project completion form after merging this branch.

All the best team,
Chris

**Pull request merged**
